packages/webview_flutter/lib/webview_flutter.dart

```diff
@@ -255,6 +256,18 @@ class WebView extends StatefulWidget {
   /// By default `debuggingEnabled` is false.
   final bool debuggingEnabled;
 
+  /// The User_Agent used for WebView requests.
+  ///
+  /// By default `userAgent` is null.
+  /// If the `userAgent` is null, the default User Agent from the Android / iOS WebView is used.
+  /// WebView User Agent for Android (https://developer.chrome.com/multidevice/user-agent)
+  ///
+  /// When the WebView is rebuild with a new User Agent, the page reloads and the request uses the new User Agent.
+  ///
+  /// When [WebViewController.goBack] is called after changing the User Agent, the old User Agent is used.
```

Comment:
Suggestion:

When [WebViewController.goBack] is called after changing `userAgent` the previous `userAgent` value is used until the page is reloaded.